src/frontend/dsl/sc.rs

```diff
@@ -120,3 +121,52 @@ where
 
     ctx.compile()
 }
+
+#[cfg(test)]
+mod tests {
+    use halo2curves::bn256::Fr;
+
+    use crate::plonkish::compiler::{
+        cell_manager::SingleRowCellManager, config, step_selector::SimpleStepSelectorBuilder,
+    };
+
+    use super::*;
+
+    #[test]
+    fn test_super_circuit_context_default() {
+        let ctx = SuperCircuitContext::<Fr, ()>::default();
+
+        assert_eq!(
+            format!("{:#?}", ctx.super_circuit),
+            format!("{:#?}", SuperCircuit::<Fr, ()>::default())
+        );
+        assert_eq!(
+            format!("{:#?}", ctx.sub_circuit_phase1),
+            format!("{:#?}", Vec::<CompilationUnit<Fr>>::default())
+        );
+        assert_eq!(ctx.sub_circuit_phase1.len(), 0);
+        assert_eq!(
+            format!("{:#?}", ctx.tables),
+            format!("{:#?}", LookupTableRegistry::<Fr>::default())
+        );
+    }
+
+    #[test]
+    fn test_super_circuit_context_sub_circuit() {
```

Comment:
@rutefig you should add tests that test all the methods, like sub_circuit_with_ast, mapping and compile. They executed with some data that makes sense and then test that the result make sense. If some methods only make sense after executing others, execute them in that order